apps/freeablo/fagui/guimanager.cpp

```diff
@@ -672,13 +672,142 @@ namespace FAGui
 
                                belt(ctx);
-                               descriptionPanel(ctx, hoverStatus.getDescription(*Engine::EngineMain::get()->mWorld->getCurrentLevel()));
+                               descriptionPanel(ctx, hoverStatus.getDescription(
+                                       *Engine::EngineMain::get()->mWorld->getCurrentLevel()));
+
+                               // Active spell
+                               int frame = (int)FAWorld::SpellData::SpellHighlightFrame::blank;
+                               auto activeSpell = mPlayer->getPlayerBehaviour()->mActiveSpell;
+
+                               if (activeSpell != FAWorld::SpellId::null) {
+                                   auto spellData = FAWorld::SpellData(activeSpell);
+                                   frame = spellData.getFrameIndex();
+                               }
+                               buttonStyle.normal = buttonStyle.hover = buttonStyle.active =
+                                       nk_style_item_image(spellIcons->getNkImage(frame));
+                               nk_layout_space_push(ctx, nk_rect(
+                                       activeSpellLeftOffset, activeSpellTopIndent,
+                                       spellIcons->getWidth(), spellIcons->getHeight()));
+                               if (nk_button_label_styled(ctx, &buttonStyle, ""))
+                                   // Toggle spell selection menu
+                                   mShowSpellSelectionMenu = !mShowSpellSelectionMenu;
+
+                                // Overlay selected hotkey symbol (F5->F8)
+                                if (activeSpell != FAWorld::SpellId::null)
+                                {
+                                    for (int j = 0; j < 4; j++)
+                                    {
+                                        if (activeSpell == mPlayer->getPlayerBehaviour()->mSpellHotkey[j])
+                                        {
+                                            int hotkeyFrame = (int)FAWorld::SpellData::SpellHighlightFrame::hotkeyF5 + j;
+                                            nk_image(ctx, spellIcons->getNkImage(hotkeyFrame));
+                                        }
+                                    }
+                                }
 
                                nk_layout_space_end(ctx);
                            },
                            false);
     }
 
+    void GuiManager::spellSelectionMenu(nk_context* ctx) {
+        // This is the menu that pops up when the spell icon on the bottom menu in clicked (or 's' hotkey).
+        // It allows the player to select the active spell, as well as configure F5->F8 hotkeys.
+
+        // TODO: Selecting a spell with a panel open then blocks input to the bottom menu buttons?
+        //  Also hover doesn't work sometimes when opening/closing other panels...
```

Comment:
I will try fix this before merging. The Nuklear GUI is playing up sometimes.
Let me know if anyone has any ideas what I might be doing wrong..

    // TODO: Selecting a spell with a panel open then blocks input to the bottom menu buttons?
    //  Also hover doesn't work sometimes when opening/closing other panels...

Reply:
nvm, just needed to set spell selection menu to background (NK_WINDOW_BACKGROUND) and use nk "inactive" functions for mouse click and hover status